### PR TITLE
Rename Semgrep Cloud Platform [(SCP)] to Semgrep AppSec Platform

### PR DESCRIPTION
**Pull request by pabloest**
I intentionally excluded editing past release notes so as not to rewrite history.

### CODE REVIEW

**Comment by s-santillan** [comment]
@pabloest Thanks for this! You may have to make an additional grep for the ff patterns:

- SCP
- SCP's
- Cloud Platform

**Comment by s-santillan** [comment]
Related: https://github.com/semgrep/semgrep-app/pull/13968

**Comment by pabloest** [comment]
> @pabloest Thanks for this! You may have to make an additional grep for the ff patterns:
> 
> * SCP
> * SCP's
> * Cloud Platform
> 
> EDIT: We should time this for shortly after the product changes are live?

Thanks! I did search for SCP and also (SCP), but not Cloud Platform on its own. I’ll check again now. Yes, we can time it as you described.

**Comment by s-santillan** [comment]
Thank you! I've been looking at it and... I'm not sure how I feel about **not** abbreviating Semgrep AppSec Platform. I'll post some screenshots here, but my initial observation is it can look heavy.

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Learn how to set up Semgrep, scan your first project for security issues, and vi
 You must have Python 3.8 or later installed on the machine where the Semgrep CLI is running.
 :::
 
-1. Navigate to [Semgrep Cloud Platform (SCP)](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to [grant Semgrep the needed permissions](/deployment/checklist/#permissions) and proceed.
+1. Navigate to [Semgrep AppSec Platform (Semgrep AppSec Platform)](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to [grant Semgrep the needed permissions](/deployment/checklist/#permissions) and proceed.
```
Comment (s-santillan) [inline docs/getting-started/quickstart.md:22]:
```suggestion
1. Navigate to [Semgrep AppSec Platform (SASP)](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to [grant Semgrep the needed permissions](/deployment/checklist/#permissions) and proceed.
```
This is why humans still have to look at these things

File: docs/semgrep-appsec-platform/semgrep-api.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+slug: semgrep-api
```
Comment (s-santillan) [inline docs/semgrep-appsec-platform/semgrep-api.md:2]: An entirely new doc? Por que?

Comment (s-santillan) [reply docs/semgrep-appsec-platform/semgrep-api.md:2]: AHHH we changed the name of the folder. Hmm. I was thinking of moving most deployment-oriented features out to /deployment, thereby future-proofing them from rebrands. The /semgrep-appsec-platform will likely survive, but with fewer docs. 

I'll apply `mv` changes on this PR, is that ok?

File: docs/getting-started/quickstart.md (hunk 1 above)
Comment (pabloest) [reply docs/getting-started/quickstart.md:22]: Yes, thank you!

File: docs/semgrep-appsec-platform/semgrep-api.md (hunk 2 above)
Comment (pabloest) [reply docs/semgrep-appsec-platform/semgrep-api.md:2]: Yes, please go ahead!

File: docs/getting-started/quickstart.md (hunk 1 above)
Comment (pabloest) [reply docs/getting-started/quickstart.md:22]: @s-santillan can you add this change to the commit you’ll push to this branch, depending on our decision about it?

Comment (s-santillan) [reply docs/getting-started/quickstart.md:22]: Yep, I'll test out "Semgrep" too, it may not always work.

**Review by s-santillan — APPROVED** [review]

**Pull request merged**